processor/src/decoder/mod.rs

```diff
@@ -335,6 +353,8 @@ impl Decoder {
         let ctx = self.span_context.as_mut().expect("not in span");
         ctx.num_groups_left -= consumed_op_groups;
         ctx.group_ops_left = op_batch.groups()[0];
+
+        self.append_operation(Operation::Span);
```

Comment:
This should be `self.append_operation(Operation::Respan)`.

Reply:
Ah, sorry, fixed.